pkg/metadata/inventories/host_metadata.go

```
@@ -120,14 +120,16 @@ func getHostMetadata() *HostMetadata {
 		metadata.CPUArchitecture = platformInfo.HardwarePlatform
 	}
 
-	memoryInfo, warnings, err := memoryGet()
+	memoryInfo := memoryGet()
+	_, warnings, err = memoryInfo.AsJSON()
 	if err != nil {
 		logErrorf("failed to retrieve host memory metadata from gohai: %s", err) //nolint:errcheck
 	} else {
 		logWarnings(warnings)
 
-		metadata.MemoryTotalKb = memoryInfo.TotalBytes / 1024
-		metadata.MemorySwapTotalKb = memoryInfo.SwapTotalBytes / 1024
+		memoryTotalKb, _ := memoryInfo.TotalBytes.Value()
+		metadata.MemoryTotalKb = memoryTotalKb / 1024
+		metadata.MemorySwapTotalKb, _ = memoryInfo.SwapTotalKb.Value()
```

`Value()` returns the default value if there is an error so here we can just ignore the error and store the value directly

---

I think you should add a comment that explains this behavior.